apps/src/block_utils.js

```diff
@@ -429,6 +429,30 @@ const DUMMY_INPUT = 'dummy';
 const STATEMENT_INPUT = 'statement';
 const FIELD_INPUT = 'field';
 
+/**
+ * Splits a blockText into labelled inputs, each match will a label followed by
+ * an input. The label is an arbitrary (possibly empty) string. The input is
+ * either a real named input like '{VALUE}', a newline, or if it matched a
+ * trailing label, nothing.
+ */
+const LABELED_INPUTS_REGEX = /.*?({[^}]*}|\n|$)/gm;
```

Comment:
Why is the `.*?` non-greedy match needed at the beginning?

Reply:
That matches the label for the input. If it's greedy, it will just consume everything between the beginning of the string and the last input.

It might be clearer to use `[^{}]*` to match the label, I could go either way.